src/argtypes.c

```diff
@@ -307,7 +307,7 @@ TypeTuple *toArgTypes(Type *t)
         void visit(TypeStruct *t)
         {
             //printf("TypeStruct::toArgTypes() %s\n", t->toChars());
-            if (!t->sym->isPOD())
+            if (!t->sym->isPOD() || t->sym->fields.dim == 0)
```

Comment:
What if the struct has a nested `this`? Might need to check `t->sym->isNested()`?